### PR TITLE
feat: allow custom in blocks

### PR DESCRIPTION
**Pull request by tyteen4a03**
## Description

Allow storing custom data in block types, in line with other types in Payload.

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [ ] Chore (non-breaking change which does not add functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Change to the [templates](https://github.com/payloadcms/payload/tree/main/templates) directory (does not affect core functionality)
- [ ] Change to the [examples](https://github.com/payloadcms/payload/tree/main/examples) directory (does not affect core functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by tyteen4a03** [comment]
I've only added the change to the type; I'm not sure if we want doc changes / tests for this?

File: packages/payload/src/fields/config/types.ts [hunk 1]
```diff
@@ -577,7 +577,9 @@ export type Block = {
    */
   interfaceName?: string
   labels?: Labels
-  slug: string
+  slug: string,
```
Comment (GeorgeHulpoi) [inline packages/payload/src/fields/config/types.ts:580]: Remove ','

**Comment by tyteen4a03** [comment]
Added some tests and docs changes just now.

File: test/config/int.spec.ts [hunk 2]
```diff
@@ -38,10 +39,17 @@ describe('Config', () => {
 
     it('allows a custom field in collection fields', () => {
       const [collection] = payload.config.collections
-      const [field] = collection.fields
+      const [field,] = collection.fields
```
Comment (GeorgeHulpoi) [inline test/config/int.spec.ts:42]: why ','?

File: packages/payload/src/fields/config/types.ts (hunk 1 above)
Comment (GeorgeHulpoi) [inline packages/payload/src/fields/config/types.ts:580]: why ','?

File: test/config/int.spec.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import payload from '../../packages/payload/src'
 import { initPayloadTest } from '../helpers/configHelpers'
+import { BlockField } from "payload/dist/fields/config/types";
```
Comment (GeorgeHulpoi) [inline test/config/int.spec.ts:3]: from 'payload/types'

File: test/config/payload-types.ts
Comment (GeorgeHulpoi) [inline test/config/payload-types.ts:1]: This file should be added in commit?

**Comment by DanRibbens** [comment]
Good feedback @GeorgeHulpoi. @tyteen4a03 are you able to open another PR to address these minor changes?

File: test/config/payload-types.ts
Comment (tyteen4a03) [reply test/config/payload-types.ts:1]: Yes, `payload-types.ts` should be generated for each of the tests. See other examples like https://github.com/payloadcms/payload/blob/main/test/collections-rest/payload-types.ts

**Pull request merged**